dask/array/core.py

```diff
@@ -889,65 +889,61 @@ def store(sources, targets, lock=True, regions=None, compute=True,
     sources_dsk = sharedict.merge(*[e.__dask_graph__() for e in sources])
     sources_dsk = Array.__dask_optimize__(
         sources_dsk,
-        [e.__dask_keys__() for e in sources]
+        list(core.flatten([e.__dask_keys__() for e in sources]))
     )
 
-    tgt_dsks = []
-    store_keys = []
-    store_dsks = []
-    if return_stored:
-        load_names = []
-        load_dsks = []
-    for tgt, src, reg in zip(targets, sources, regions):
-        # if out is a delayed object update dictionary accordingly
+    # Optimize all targets together
+    targets_keys = []
+    targets_dsk = []
+    for e in targets:
         try:
-            each_tgt_dsk = {}
-            each_tgt_dsk.update(tgt.dask)
-            tgt = tgt.key
+            targets_keys.extend(e.__dask_keys__())
+            targets_dsk.append(e.__dask_graph__())
         except AttributeError:
-            each_tgt_dsk = {}
+            pass
+
+    targets_dsk = sharedict.merge(*targets_dsk)
+    targets_dsk = Delayed.__dask_optimize__(targets_dsk, targets_keys)
 
+    load_stored = (return_stored and not compute)
```

Comment:
nit: these parenthesis are unnecessary

Reply:
I know they aren't necessary, but it feels more readable to me. If there are strong objections, can change though.